File: icons/generate-icons.js
// Run once: node icons/generate-icons.js
// Generates icon-192.png and icon-512.png from the SVG using canvas-like drawing.
// Uses only Node.js built-ins — no npm required.

const fs   = require('fs');
const path = require('path');

// We'll write minimal PNG files by encoding a simple canvas drawing.
// Uses the 'canvas' npm package if available, otherwise falls back to
// writing pre-encoded minimal PNGs.

// Try to use the @napi-rs/canvas or canvas package
let createCanvas;
try {
  ({ createCanvas } = require('@napi-rs/canvas'));
} catch (_) {
  try {
    ({ createCanvas } = require('canvas'));
  } catch (_) {
    createCanvas = null;
  }
}

if (!createCanvas) {
  console.log('No canvas module found — generating icons via Python fallback.');
  console.log('Run: python icons/generate-icons.py');
  process.exit(0);
}

function drawIcon(size) {
  const canvas = createCanvas(size, size);
  const ctx    = canvas.getContext('2d');
  const cx = size / 2, cy = size / 2, r = size / 2;

  // Background
  ctx.fillStyle = '#1e1e1e';
  ctx.beginPath();
  ctx.arc(cx, cy, r, 0, Math.PI * 2);
  ctx.fill();

  // Orange ring
  ctx.strokeStyle = '#e67e22';
  ctx.lineWidth   = size * 0.035;
  ctx.beginPath();
  ctx.arc(cx, cy, r * 0.86, 0, Math.PI * 2);
  ctx.stroke();

  // Clay target disc
  ctx.save();
  ctx.translate(cx, cy * 0.7);
  ctx.scale(1, 0.3);
  ctx.fillStyle = '#e67e22';
  ctx.beginPath();
  ctx.arc(0, 0, size * 0.14, 0, Math.PI * 2);
  ctx.fill();
  ctx.fillStyle = '#c0392b';
  ctx.beginPath();
  ctx.arc(0, 0, size * 0.1, 0, Math.PI * 2);
  ctx.fill();
  ctx.restore();

  // Gun barrel
  const barrelY = cy + size * 0.1;
  ctx.fillStyle = '#a0a0a0';
  ctx.beginPath();
  ctx.roundRect(size * 0.18, barrelY - size * 0.02, size * 0.45, size * 0.04, size * 0.02);
  ctx.fill();

  // Stock
  ctx.fillStyle = '#5a3e2b';
  ctx.beginPath();
  ctx.moveTo(size * 0.18, barrelY - size * 0.02);
  ctx.quadraticCurveTo(size * 0.1, barrelY + size * 0.04, size * 0.14, barrelY + size * 0.1);
  ctx.lineTo(size * 0.26, barrelY + size * 0.02);
  ctx.closePath();
  ctx.fill();

  return canvas.toBuffer('image/png');
}

for (const size of [192, 512]) {
  const buf  = drawIcon(size);
  const dest = path.join(__dirname, `icon-${size}.png`);
  fs.writeFileSync(dest, buf);
  console.log(`Written: ${dest}`);
}

File: icons/generate-icons.py
"""
Generate icon-192.png and icon-512.png for the Trap N Skeet PWA.
Run: python icons/generate-icons.py
Requires: pip install Pillow
"""
import math, struct, zlib, os, sys

try:
    from PIL import Image, ImageDraw
except ImportError:
    print("Pillow not found. Installing...")
    import subprocess
    subprocess.check_call([sys.executable, "-m", "pip", "install", "Pillow", "--quiet"])
    from PIL import Image, ImageDraw

OUT_DIR = os.path.dirname(os.path.abspath(__file__))

def hex_to_rgb(h):
    h = h.lstrip('#')
    return tuple(int(h[i:i+2], 16) for i in (0, 2, 4))

def draw_icon(size):
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    d   = ImageDraw.Draw(img)

    cx, cy = size // 2, size // 2

    # Background circle
    d.ellipse([0, 0, size-1, size-1], fill=(*hex_to_rgb('#1e1e1e'), 255))

    # Orange ring
    ring_r     = int(size * 0.86 / 2)
    ring_w     = max(3, int(size * 0.035))
    ring_box   = [cx - ring_r, cy - ring_r, cx + ring_r, cy + ring_r]
    d.ellipse(ring_box, outline=(*hex_to_rgb('#e67e22'), 255), width=ring_w)

    # Clay target — outer orange ellipse
    tx, ty = cx, int(cy * 0.7)
    rw, rh = int(size * 0.14), int(size * 0.042)
    d.ellipse([tx-rw, ty-rh, tx+rw, ty+rh], fill=(*hex_to_rgb('#e67e22'), 255))
    # Inner red ring
    rw2, rh2 = int(size * 0.10), int(size * 0.03)
    d.ellipse([tx-rw2, ty-rh2, tx+rw2, ty+rh2], fill=(*hex_to_rgb('#c0392b'), 255))
    # Center orange dot
    rw3, rh3 = int(size * 0.06), int(size * 0.018)
    d.ellipse([tx-rw3, ty-rh3, tx+rw3, ty+rh3], fill=(*hex_to_rgb('#e67e22'), 220))

    # Gun barrel
    bx1 = int(size * 0.18)
    bx2 = int(size * 0.63)
    by  = int(cy + size * 0.10)
    bh  = max(4, int(size * 0.04))
    d.rounded_rectangle([bx1, by - bh//2, bx2, by + bh//2],
                         radius=bh//2, fill=(*hex_to_rgb('#a0a0a0'), 255))

    # Stock triangle
    stock_pts = [
        (bx1, by - bh//2),
        (int(size * 0.10), by + int(size * 0.04)),
        (int(size * 0.14), by + int(size * 0.10)),
        (int(size * 0.26), by + int(size * 0.02)),
    ]
    d.polygon(stock_pts, fill=(*hex_to_rgb('#5a3e2b'), 255))

    # Dashed sight line from barrel tip to clay target
    x1, y1 = bx2, by
    x2, y2 = tx, ty + rh
    # Draw dashes
    steps  = 12
    for i in range(0, steps, 2):
        px1 = int(x1 + (x2 - x1) * i / steps)
        py1 = int(y1 + (y2 - y1) * i / steps)
        px2 = int(x1 + (x2 - x1) * (i+1) / steps)
        py2 = int(y1 + (y2 - y1) * (i+1) / steps)
        d.line([(px1, py1), (px2, py2)],
               fill=(*hex_to_rgb('#e67e22'), 180), width=max(2, size//100))

    return img

for size in [192, 512]:
    img  = draw_icon(size)
    dest = os.path.join(OUT_DIR, f'icon-{size}.png')
    img.save(dest, 'PNG')
    print(f'Written: {dest}')

print('Done!')
